second task after fixes and third task

VitaliyPrihotskiy opened the pull request.

Commit:
second task after fixes

IvanOkhrimenko commented on my-app/my-app/src/app/app.module.ts:
```diff
@@ -0,0 +1,30 @@
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { CoreModule } from './core/core.module';
+
+
```
delete blank line

IvanOkhrimenko commented on my-app/my-app/src/app/core/core.module.ts:
```diff
@@ -0,0 +1,40 @@
+import { CoursesComponent } from './courses/courses.component';
+import { CourseComponent } from './course/course.component';
+
+
```
blank lines

IvanOkhrimenko commented on my-app/my-app/src/app/app.module.ts:
```diff
@@ -0,0 +1,30 @@
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+
```
delete

IvanOkhrimenko commented on my-app/my-app/src/app/core/course/course.component.ts:
```diff
@@ -0,0 +1,18 @@
+@Component({
+  selector: 'app-course',
+  templateUrl: './course.component.html',
+  styleUrls: ['./course.component.scss']
```
detect changes strategy

IvanOkhrimenko commented on my-app/my-app/src/app/core/course/course.component.ts:
```diff
@@ -0,0 +1,18 @@
+  @Input() course: Course | null = null;
+  @Output() deleteItemEvent = new EventEmitter<number>();
+
+  constructor() { }
```
delete empty methods

IvanOkhrimenko commented on my-app/my-app/src/app/core/course/course.component.ts:
```diff
@@ -0,0 +1,18 @@
+
+  constructor() { }
+
+  delete(id:number){
```
return function type

IvanOkhrimenko commented on my-app/my-app/src/app/core/courses/courses.component.spec.ts:
```diff
@@ -0,0 +1,23 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
```
blank line

Commit:
modified:   my-app/package.json
	modified:   my-app/src/app/app.module.ts
	modified:   my-app/src/app/core/core.module.ts

Commit:
third task

IvanOkhrimenko commented on my-app/src/app/constants/courses-list.ts:
```diff
@@ -0,0 +1,31 @@
+import { Course } from "../models/models";
+
+export function getCourses(amount: number): Course[] {
```
я пропонував просто зробити константу з курсами, без генератора, ти ж для тестів не зможеш переюзати саме цю функцію

IvanOkhrimenko commented on my-app/src/app/core/courses/courses.component.ts:
```diff
@@ -1,59 +1,59 @@
   count = 3;
   filteredShownCourses: Course[] = [];
-  constructor() { }
+  constructor(private cd:ChangeDetectorRef) { }
```
private readonly cd:ChangeDetectorRef

IvanOkhrimenko commented on my-app/src/app/core/courses/courses.component.ts:
```diff
@@ -1,59 +1,59 @@
     let itemList = this.courses.slice();
     input = input.toLowerCase();
     let twoMatches = itemList.filter(e => e.header.toLowerCase().includes(input) && e.description.toLowerCase().includes(input));
+    console.log(twoMatches)
```
delete consoles

IvanOkhrimenko commented on my-app/src/app/core/courses/courses.component.ts:
```diff
@@ -1,59 +1,59 @@
+    console.log(titleMatches.length)
     let summaryMatches = itemList.filter(e => e.description.toLowerCase().includes(input) && !twoMatches.includes(e))
     this.filteredCourses = [...twoMatches, ...titleMatches, ...summaryMatches];
+    this.filteredShownCourses = this.filteredCourses.slice().splice(0, this.count);
```
better use filter than splice

IvanOkhrimenko commented on my-app/src/app/core/courses/courses.component.ts:
```diff
@@ -1,59 +1,59 @@
 export class CoursesComponent implements OnInit {
   courses: Course[] = [];
   filteredCourses: Course[] = [];
   count = 3;
```
rename it, pageCount, or something like that

IvanOkhrimenko commented on my-app/src/app/core/courses/courses.component.ts:
```diff
@@ -1,59 +1,59 @@
   }
 
-  refresh() {
+  refresh(): void {
```
please, use more consistent naming, for this case, in my opinion, better to use resetFilteredCourses, or something like that

IvanOkhrimenko commented on my-app/src/app/core/login/login.component.ts:
```diff
@@ -1,10 +1,12 @@
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.scss']
+  styleUrls: ['./login.component.scss'],
+    providers: [FormBuilder],
```
do you need to provide it here?

IvanOkhrimenko commented on my-app/src/app/core/new-course/new-course.component.ts:
```diff
@@ -1,18 +1,20 @@
   templateUrl: './new-course.component.html',
-  styleUrls: ['./new-course.component.scss']
+  styleUrls: ['./new-course.component.scss'],
+  providers: [FormBuilder],
```
same

IvanOkhrimenko commented on my-app/src/app/core/new-course/new-course.component.ts:
```diff
@@ -1,18 +1,20 @@
   dateControl = new FormControl('', Validators.required);
-  authorsControl= new FormControl('', Validators.required);
+  authorsControl = new FormControl('', Validators.required);
   course: FormGroup = this._formBuilder.group({
```
as do you use FormControl, lest use new FormGroup

IvanOkhrimenko commented on my-app/src/app/core/courses/courses.component.spec.ts:
```diff
@@ -20,4 +21,35 @@ describe('CoursesComponent', () => {
+    expect(component.courses.length).toBe(lenghtBefore-1);
+  });
+
+  it('should filter courses to 1 course', () => {
```
should filter courses to course with name 'Video Course 1.Name tag'

IvanOkhrimenko commented on my-app/src/app/core/courses/courses.component.spec.ts:
```diff
@@ -20,4 +21,35 @@ describe('CoursesComponent', () => {
+  it('should filter courses to 1 course', () => {
+    component.filter("Video Course 1.Name tag");
+    fixture.detectChanges();
+    expect(component.filteredShownCourses.length).toBe(1);
```
lets check that filteredShownCourses should contain course with name Video Course 1.Name tag
